prow/moonraker/client.go

```diff
@@ -95,6 +95,10 @@ func (c *Client) do(method, path string, payload []byte, params map[string]strin
 		q.Set(key, val)
 	}
 	req.URL.RawQuery = q.Encode()
+	timeout := c.configAgent.Config().Moonraker.ClientTimeout.Duration
+	c.Lock()
+	c.httpClient.Timeout = timeout
+	c.Unlock()
```

Comment:
The locking doesn't seem to protect potential concurrent reads of Timeout.

Reply:
According to https://pkg.go.dev/net/http#Client the Clients are safe for use by multiple goroutines.

Reply:
As discussed, the issue is the potential for another goroutine to read httpClient.Timeout while it is being written here, without any locking. To avoid this, the simplest solution is to replace c.httpClient with a new client when the configuration changes. In this solution, the http.Client is never modified once created.

Reply:
I did my own version (without the helper method) then saw this comment. I've since pushed up a version using a helper method. I don't understand why we need the intermediate `hc` variable as in your example so I left it out.

Reply:
To clarify, hc is needed in my example implementation in order to call hc.Do() later. This is because c.httpClient must only be accessed while locked.